components/director/internal/domain/apptemplate/resolver.go

```diff
@@ -1060,6 +1060,16 @@ func (r *Resolver) prepareCertSubjectMapping(ctx context.Context, appTemplateID,
 		}
 	}
 
+	exists, err := r.certSubjectMappingSvc.ExistsBySubject(ctx, subject)
+	if err != nil {
+		return errors.Wrapf(err, "while checking if a certificate subject mapping exists with a subject: %s", subject)
+	}
+
+	if exists {
+		log.C(ctx).Debug("Subject is already allow-listed. Skipping certificate subject mapping creation.")
+		return nil
+	}
+
```

Comment:
Will this be enough as a check? because in the repo layer, we'll compare both subjects as strings using SQL where clause with "subject = X"? I'm not sure what will be the results if we have two subjects separated with a comma where in one of them we have spaces and in the other, we don't? Or if the certificate subject properties(CN, O, L, etc..) are in a different order?

We could get into a situation where we have two subjects with the same properties but with different consumer_type, access_levels, etc..?